pkg/controller/datavolume-controller.go

```diff
@@ -532,6 +532,16 @@ func (r *DatavolumeReconciler) selectCloneStrategy(datavolume *cdiv1.DataVolume,
 		return NoClone, nil
 	}
 
+	storageClass, err := GetStorageClassByName(r.client, pvcSpec.StorageClassName)
+
+	if err != nil {
+		return NoClone, err
+	}
+
+	if strings.HasPrefix(storageClass.Provisioner, "openshift-storage") && storageClass.Parameters["encrypted"] == "true" {
```

Comment:
I would prefer that the recommended cloning strategy actually appears in the StorageProfile status.  This means that this logic should be part of the StorageProfile controller's reconcile loop.  If we do it there, then there is no need to special case it here.

Reply:
I agree. StorageProfile has a field called cloneStrategy. And it looks like this logic fits nicely to storageProfile. 

That way an end user can see which strategy will be applied by checking the storageprofile status.  Also the cloning in datavolume will remain simple.

Reply:
updated with new changes